app/views/layouts/application.html.erb

```diff
@@ -9,11 +9,44 @@
   </head>
 
   <body>
+    <% if session[:user_id] != nil %>
+      <% user = User.find(session[:user_id]) %>
+    <% end %>
     <nav class = "topnav">
-      <%= link_to "All Merchants", "/merchants"%>
-      <%= link_to "All Items", "/items"%>
-      <%= link_to "Cart: #{cart.total_items}", "/cart" %>
-      <%= link_to "Register", "/register" %>
+      <% if user == nil %>
+        <%= link_to "Home", "/" %>
+        <%= link_to "All Merchants", "/merchants"%>
+        <%= link_to "All Items", "/items"%>
+        <%= link_to "Cart: #{cart.total_items}", "/cart" %>
+        <%= link_to "Register", "/register" %>
+        <%= link_to "Log in", "/login" %>
+      <% elsif user.role == "default" %>
```

Comment:
All these `user.role == "role"` conditions could refactored as `user.role?` for example `user.admin?`. That a method that comes with the enum we used in the User model.